### PR TITLE
Rebuild of previous pr typingLessons for gitleaks

### PR DESCRIPTION
**Pull request by kevincasey1222**
Hi @mknoedel !

So here is a cleanup rebuild of the previous PR that doesn't have any fake secrets in the git history.

The problem that is keeping it from passing the tests is something to do with authentication and the Polly recordings. The tests all pass with a local var.

The good news is that we're on the latest version of the @octokit modules now. 

--Kevin

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -119,62 +119,83 @@ export class APIClient {
   }
 
   public async setupAccountClient(): Promise<void> {
+    const authen: boolean = !(
```
Comment (mknoedel) [inline src/client.ts:122]: I think something like `shouldAuthenticate` would be a bit more clear.

File: src/client.ts [hunk 2]
```diff
@@ -119,62 +119,83 @@ export class APIClient {
...
+      this.config.githubAppPrivateKey === 'donotauthenticate'
+    ); //a hook for testing to skip certain authens if needed, because Polly isn't capturing all the authen
+
+    const installationId = Number(this.config.installationId);
```
Comment (mknoedel) [inline src/client.ts:126]: Probably want to validate with isNaN here

File: src/client.ts [hunk 3]
```diff
@@ -119,62 +119,83 @@ export class APIClient {
...
+    const appClient = createGitHubAppClient(this.config, this.logger);
+    let myToken: string = '[REDACTED]';
```
Comment (mknoedel) [inline src/client.ts:128]: This initialization pattern is a bit confusing. Could this be written in such a way where it is more obvious that the test environment has a token of '[REDACTED]' and so on?

Comment (mknoedel) [reply src/client.ts:128]: As this is a bandaid anyway, perhaps an `isTest` checks an environment variable might suffice for now until we cut over.

Comment (kevincasey1222) [reply src/client.ts:128]: Band aid removed

File: src/client.ts (hunk 2 above)
Comment (kevincasey1222) [reply src/client.ts:126]: Done

File: src/client.ts (hunk 1 above)
Comment (kevincasey1222) [reply src/client.ts:122]: Remvoed

File: src/util/sha.ts [hunk 4]
```diff
@@ -5,3 +5,13 @@ export default function sha(text: string): string {
   sha256.update(text);
   return sha256.digest('hex').toString();
 }
+
+export function getFakeRsaKey(): string {
```
Comment (mknoedel) [inline src/util/sha.ts:9]: love it!

**Review by mknoedel — APPROVED** [review]

**Pull request merged**